### ISSUE TITLE
Jest --update-snapshot flag incorrectly set as default

### ISSUE THREAD

**Issue by automartin5000** [projen/projen#3487]
For the test task, it looks like the [default behavior](https://github.com/projen/projen/blob/d37de27fdb46af03c6acde21d2e4f5793e3b81ca/src/javascript/jest.ts#L788) is for jest to run with the --update-snapshot flag. This seems to be the case even in [projen's own builds](https://github.com/projen/projen/blob/91d58e0bb927a758c3e3004a7a69d9bbb780b108/.projen/tasks.json#L359). 

I'm new to snapshot testing, but based on some simple testing, that seems to mean that tests always pass as snapshots are just always updated and code changes never fail. Shouldn't the default be `UpdateSnapshot.NEVER`? Indeed, it seems like that was the intention of [this code](https://github.com/projen/projen/blob/91d58e0bb927a758c3e3004a7a69d9bbb780b108/src/javascript/jest.ts#L915).

**Comment by mrgrain** [comment]
There's a separate mechanism that fails PR builds if they have any mutations, including snapshot changes (or mutates the PR if you are using mutable PRs). And PRs require reviews, so this is still fine. 

A lot of folks don't agree with this approach, so we probably ought to change the default _at some point_.

---

My very personal opinion is that I like to view snapshot changes using the file system and my IDEs diff tool, not whatever jest puts on the command line. From a untrusted code standpoint, there's also no guarantee that someone has actually looked at the snapshot changes and decided they are good vs just re-run the test with `--update-snapshot`. So the only safe mechanism to see if I want this change or not is reviewing the changes.

**Comment by automartin5000** [comment]
> There's a separate mechanism that fails PR builds if they have any mutations, including snapshot changes (or mutates the PR if you are using mutable PRs). And PRs require reviews, so this is still fine.
> 
> A lot of folks don't agree with this approach, so we probably ought to change the default _at some point_.
> 
> My very personal opinion is that I like to view snapshot changes using the file system and my IDEs diff tool, not whatever jest puts on the command line. From a untrusted code standpoint, there's also no guarantee that someone has actually looked at the snapshot changes and decided they are good vs just re-run the test with `--update-snapshot`. So the only safe mechanism to see if I want this change or not is reviewing the changes. At the end of the day I am committing/pushing/merging code. It's my responsibility to make sure it's the code I want. To me the extra step of failing tests doesn't provide much benefit, but I can appreciate that it does to others.

My biggest concern with this approach is that it runs counter to standard testing workflows, at least the ones I'm used to: Write some tests, run `projen test`, push to branch. 

What happens if someone accidentally runs `projen test` locally before testing snapshots? Then they don't get feedback until the anti-tampering check fails? Then the developer has to go and parse the anti-tampering check and figure out what failed.

**Comment by mrgrain** [comment]
> What happens if someone accidentally runs projen test locally before testing snapshots? Then they don't get feedback until the anti-tampering check fails? Then the developer has to go and parse the anti-tampering check and figure out what failed.

Yeah, I understand that concern. It's totally valid, hence why I think the default should change. My only response to this is that the developer has to actively ignore the pending local changes and decide to not commit them to even get into this situation.
